orchestrator/graphql/resolvers/subscription.py

```diff
@@ -65,26 +65,34 @@
 def get_subscription_graphql_type(info: OrchestratorInfo, subscription_name: str) -> StrawberryTypeFromPydantic:
     subscription_graphql_type = info.context.graphql_models.get(subscription_name)
     if not subscription_graphql_type:
-        raise GraphQLError(message=f"No graphql type found for {subscription_name}")
+        logger.warn(message=f"No graphql type found for {subscription_name}")
+        base_type = info.context.graphql_models.get("subscription")
+        if not base_type:
+            raise GraphQLError("No subscription base type found")
+        return base_type
     return subscription_graphql_type
 
 
-def get_subscription_details(info: OrchestratorInfo, subscription: SubscriptionTable) -> SubscriptionInterface:
+async def get_subscription_details(info: OrchestratorInfo, subscription: SubscriptionTable) -> SubscriptionInterface:
+    from orchestrator.domain import SUBSCRIPTION_MODEL_REGISTRY
     from orchestrator.graphql.autoregistration import graphql_subscription_name
 
-    subscription_details = SubscriptionModel.from_subscription(subscription.subscription_id)
-    base_model = subscription_details.__base_type__ if subscription_details.__base_type__ else subscription_details
-    base_subscription_details = base_model.from_other_lifecycle(  # type: ignore
-        subscription_details, SubscriptionLifecycle.INITIAL, skip_validation=True
-    )
-    base_subscription_details.status = subscription_details.status
-    strawberry_type = get_subscription_graphql_type(info, graphql_subscription_name(base_model.__name__))  # type: ignore
-    return strawberry_type.from_pydantic(base_subscription_details)  # type:ignore
+    subscription_dict_data, _ = await get_subscription_dict(subscription.subscription_id)
+
+    domain_model_type = SUBSCRIPTION_MODEL_REGISTRY[subscription.product.name]
+    base_model = domain_model_type.__base_type__ if domain_model_type.__base_type__ else domain_model_type
+
+    subscription_name = graphql_subscription_name(base_model.__name__)
+    subscription_details = base_model.model_validate(subscription_dict_data, strict=False)
```

Comment:
https://docs.pydantic.dev/latest/concepts/models/#creating-models-without-validation

Reply:
Oh.

> In Pydantic V2, the performance gap between BaseModel.__init__ and BaseModel.model_construct has been narrowed considerably. For simple models, calling BaseModel.__init__ may even be faster. If you are using [model_construct()](https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel.model_construct) for performance reasons, you may want to profile your use case before assuming that [model_construct()](https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel.model_construct) is faster.

Nvm then, maybe worth a shot in the future though if some subscriptions seem particularly slow to resolve